src/runtime/refiner.ts

```diff
@@ -0,0 +1,140 @@
+/**
+ * @license
+ * Copyright (c) 2019 Google Inc. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {Refinement, SchemaPrimitiveType} from './manifest-ast-nodes.js';
+import {Dictionary} from './hot.js';
+import {assert} from '../platform/assert-node.js';
+
+export class Refiner {
+    static refinementString(refinement) : string {
```

Comment:
document this please; why is it a static method on Refiner rather than a toString() member method?